### PR TITLE
chore: organize per-browser dependencies

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -142,9 +142,9 @@ jobs:
     - uses: actions/setup-node@v1
       with:
         node-version: 10
-    - uses: microsoft/playwright-github-action@v1
     - run: npm ci
     - run: npm run build
+    - run: node lib/cli/cli install
```
Comment (yury-s) [inline .github/workflows/tests.yml:147]: Let's update gha to use cli?

**Review by yury-s — APPROVED** [review]

File: src/install/installDeps.ts [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    ...deps.chromium,
+    ...deps.firefox,
+    ...deps.webkit
+  ].join(' '));
+
+  // - `ffmpeg`: For video playback in Firefox
+  commands.push('apt-get install -y ffmpeg');
+  // For headful execution
+  commands.push('apt-get install -y xvfb');
+  const child = childProcess.spawn('sudo', ['--', 'sh', '-c', `${commands.join('; ')}`], { stdio: 'inherit' });
```
Comment (dgozman) [inline src/install/installDeps.ts:265]: Why not childProcess.execSync() instead?

File: src/install/installDeps.ts [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+  await new Promise(f => child.on('exit', f));
+}
+
+
+async function getUbuntuVersion(): Promise<string> {
```
Comment (dgozman) [inline src/install/installDeps.ts:270]: We have src/utils/ubuntuVersion.ts

File: src/install/installDeps.ts [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+  const commands: string[] = [];
+  commands.push('apt-get update');
+  const ubuntuVersion = await getUbuntuVersion();
+  let deps: any = [];
+  if (ubuntuVersion === '18.04')
+    deps = DEPENDENCIES['ubuntu18.04'];
+  else if (ubuntuVersion === '20.04')
+    deps = DEPENDENCIES['ubuntu20.04'];
+  else
+    console.warn('Cannot install dependencies for this linux distribution!');
```
Comment (dgozman) [inline src/install/installDeps.ts:253]: Perhaps early return here? Do all linux distributions have apt-get that we call below?

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -64,9 +64,9 @@ jobs:
...
```
Comment (dgozman) [inline .github/workflows/tests.yml:69]: Let's try `install ${{ matrix.browser }}` right away?

File: src/install/installDeps.ts [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+    return;
+  }
+  console.log('Installing Ubuntu dependencies...');
+  commands.push(['apt-get', 'install', '-y', '--no-install-recommends',
+    ...deps.chromium,
+    ...deps.firefox,
+    ...deps.webkit
+  ].join(' '));
+
+  // - `ffmpeg`: For video playback in Firefox
```
Comment (mxschmitt) [inline src/install/installDeps.ts:263]: let's maybe have a single apt-get install call?

**Review by mxschmitt — APPROVED** [review]

File: src/install/installDeps.ts (hunk 2 above)
Comment (pavelfeldman) [reply src/install/installDeps.ts:264]: To show progress...

File: .github/workflows/tests.yml (hunk 1 above)
Comment (pavelfeldman) [reply .github/workflows/tests.yml:147]: We'll probably sunset it

File: .github/workflows/tests.yml (hunk 5 above)
Comment (pavelfeldman) [reply .github/workflows/tests.yml:69]: Can't do - we need Chromium for recorder.

File: src/install/installDeps.ts (hunk 6 above)
Comment (pavelfeldman) [reply src/install/installDeps.ts:260]: Sure, I can try doing this. I hope --no-install-recommends won't break things for ffmpeg.

File: src/cli/cli.ts [hunk 7]
```diff
@@ -82,20 +83,34 @@ program
...
+    });
+
+program
+    .command('install-deps [browserType...]')
+    .description('install dependencies necessary to run browser')
+    .action(async function(browserType) {
+      try {
+        await installDeps(browserType);
+      } catch (e) {
         console.log(`Failed to install browsers\n${e}`);
```
Comment (pavelfeldman) [inline src/cli/cli.ts:111]: I'll fix the message

File: .github/workflows/tests.yml [hunk 8]
```diff
@@ -29,9 +29,9 @@ jobs:
     - uses: actions/setup-node@v1
       with:
         node-version: 10
-    - uses: microsoft/playwright-github-action@v1
     - run: npm ci
     - run: npm run build
+    - run: node lib/cli/cli install-deps ${{ matrix.browser }} chromium
```
Comment (yury-s) [inline .github/workflows/tests.yml:34]: remove `chromium`

File: .github/workflows/tests.yml [hunk 9]
```diff
@@ -64,9 +64,9 @@ jobs:
...
```
Comment (yury-s) [inline .github/workflows/tests.yml:69]: ditto

File: .github/workflows/tests.yml [hunk 10]
```diff
@@ -142,9 +142,9 @@ jobs:
...
```
Comment (yury-s) [inline .github/workflows/tests.yml:147]: ditto

File: src/server/validateDependencies.ts [hunk 11]
```diff
@@ -297,16 +297,16 @@ const LIBRARY_TO_PACKAGE_NAME_UBUNTU_18_04: { [s: string]: string} = {
   'libglib-2.0.so.0': 'libglib2.0-0',
   'libgmodule-2.0.so.0': 'libglib2.0-0',
   'libgobject-2.0.so.0': 'libglib2.0-0',
-  'libgstapp-1.0.so.0': 'libgstreamer-plugins-base1.0-0',
-  'libgstaudio-1.0.so.0': 'libgstreamer-plugins-base1.0-0',
+  'libgstapp-1.0.so.0': 'gstreamer1.0-plugins-base',
```
Comment (yury-s) [inline src/server/validateDependencies.ts:300]: See the comment above:
```
// This list is generted with the following program:
// ./utils/linux-browser-dependencies/run.sh ubuntu:18.04
```
should this part of mapping go there?

**Review by dgozman — APPROVED** [review]

**Pull request merged**